handlers/ip-handler.js

```diff
@@ -0,0 +1,121 @@
+'use strict'
+const fs = require('fs')
+const path = require('path')
+
+const DeveloperHandler = require('../handlers/developers-handler.js')
+const processUpload = require('../src/process-upload')
+
+const IPHandler = {
+  /**
+   * Starts the e2e tests once a developers credentials are verified or travis is running the tests
+   * @param {Object} req - The express formatted object containing the files and the tarball
+   * @param {Object} res - The response object containing any errors produced during the testing process
+   */
+  startTest: function (req, res) {
+    const filename = req.files.tarball.name
+    const entryPoint = req.body['entry-point'] || 'demo'
+    const testsFolder = req.body['tests-folder'] || 'tests/e2e'
+    processUpload.newFile(filename, entryPoint, testsFolder, res)
+  },
+  /**
+   * Checks the IP of the incoming tarball
+   * @param {Object} req - The express formatted object containing the files and credentials
+   * @returns {Promise} A promise that will either resolve if the credentials passed are authentic
+   * or reject if there are errors
+   */
+  checkIP: function (req) {
+    return new Promise((resolve, reject) => {
+      const ip = req.headers['x-forwarded-for'].toString()
+      if (!ip) {
+        reject(`Please set headers for proxy connections using nginx!\n
+        A helpful article on setting this up: https://www.digitalocean.com/community/tutorials/
+        understanding-nginx-http-proxying-load-balancing-buffering-and-caching`)
+      }
+      let fileContents
+      try {
+        fileContents = JSON.parse(fs.readFileSync(path.join(__dirname, '..', 'src', 'info.json'))).ip
```

Comment:
I'm confused why we're loading a file here rather than just looking at the headers from the request?